### PR TITLE
PARQUET-1904: [C++] Export file_offset in RowGroupMetaData

### PR DESCRIPTION
**Pull request by scober**
Export the file_offset field (which already exists in RowGroupMetaDataImpl) in RowGroupMetaData.

### CODE REVIEW

**Comment by pitrou** [comment]
@emkornfield Do you think this is ok to add?

**Comment by emkornfield** [comment]
Since this is simply a proxy object around the underlying metadata I think it is OK to expose but should have some documentation around the value of this field if the offset isn't present (it looks like this field is [optional](https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L822), but in practice I imagine it must be required to properly parse a file?).  It would be nice to use the wording from the parquet specificaiton

**Comment by scober** [comment]
When you say documentation I assume you mean a comment near the method definition?
And it looks the default value of this field is 0? (https://github.com/apache/arrow/blob/d02e16646cb3ee2a568dfeeaa68a0f75986592c8/cpp/src/parquet/metadata.cc#L1249)
And when you say the wording from the parquet specification do you mean the comment in the thrift definition?

**Comment by emkornfield** [comment]
> When you say documentation I assume you mean a comment near the method definition?

Yes the comment for the method definition

> And it looks the default value of this field is 0?

Yes, I'm not sure if it breaks anything to make this -1 or something obviously wrong (0 should be wrong because I think all parquet files need to start with 'PAR1'

> And when you say the wording from the parquet specification do you mean the comment in the thrift definition?

Yes:
```
Byte offset from beginning of file to first page (data or dictionary) in this row group
```

**Comment by scober** [comment]
I updated the comments for the method declaration to address your documentation concerns. I did not change the default value of `file_offset` because I agree that 0 is wrong (though less obviously wrong than -1) and I did not want to chase down the unintended consequences of changing that.

File: cpp/src/parquet/metadata.h [hunk 1]
```diff
@@ -210,6 +210,12 @@ class PARQUET_EXPORT RowGroupMetaData {
 
   /// \brief Total byte size of all the uncompressed column data in this row group.
   int64_t total_byte_size() const;
+
+  // The file_offset field that this method exposes is optional. This method
```
Comment (emkornfield) [inline cpp/src/parquet/metadata.h:214]: Please move this below \brief with a new line and use triple slash '///'

Comment (scober) [reply cpp/src/parquet/metadata.h]: Sure

**Comment by emkornfield** [comment]
+1 thank you @scober

**Pull request closed without merging**